x-pack/legacy/plugins/spaces/public/management/management_service.test.ts

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ManagementService } from '.';
+
+describe('ManagementService', () => {
+  describe('#start', () => {
+    it('registers the spaces management page under the kibana section', () => {
+      const mockKibanaSection = {
+        hasItem: jest.fn().mockReturnValue(false),
+        register: jest.fn(),
+      };
+
+      const managementStart = {
+        legacy: {
+          getSection: jest.fn().mockReturnValue(mockKibanaSection),
+        },
+      };
+
+      const deps = {
+        managementStart,
+      };
+
+      const service = new ManagementService();
+      service.start(deps);
+
+      expect(deps.managementStart.legacy.getSection).toHaveBeenCalledTimes(1);
+      expect(deps.managementStart.legacy.getSection).toHaveBeenCalledWith('kibana');
+
+      expect(mockKibanaSection.register).toHaveBeenCalledTimes(1);
+      expect(mockKibanaSection.register).toHaveBeenCalledWith('spaces', {
+        name: 'spacesManagementLink',
+        order: 10,
+        display: 'Spaces',
+        url: `#/management/spaces/list`,
+      });
+    });
+
+    it('will not register the spaces management page twice', () => {
+      const mockKibanaSection = {
+        hasItem: jest.fn().mockReturnValue(true),
+        register: jest.fn(),
+      };
+
+      const managementStart = {
+        legacy: {
+          getSection: jest.fn().mockReturnValue(mockKibanaSection),
+        },
+      };
+
+      const deps = {
+        managementStart,
+      };
+
+      const service = new ManagementService();
+      service.start(deps);
+
+      expect(deps.managementStart.legacy.getSection).toHaveBeenCalledTimes(1);
+      expect(mockKibanaSection.register).toHaveBeenCalledTimes(0);
+    });
+
+    it('will not register the spaces management page if the kibana section is missing', () => {
+      const managementStart = {
+        legacy: {
+          getSection: jest.fn().mockReturnValue(undefined),
+        },
+      };
+
+      const deps = {
+        managementStart,
+      };
+
+      const service = new ManagementService();
+      service.start(deps);
+
+      expect(deps.managementStart.legacy.getSection).toHaveBeenCalledTimes(1);
```

Comment:
__question:__ shouldn't we check here that `register` wasn't called instead?

Reply:
good catch!

Reply:
Actually, since the Kibana section isn't available on this scenario, there isn't a `register` function to assert against.